### PR TITLE
correct genSalt error in user update

### PR DESCRIPTION
**Pull request by NP-SON-250**
## What doe this pull request do?
1. Correct error on user update
2. Change default link for postImage on cloudinary

### CODE REVIEW

File: src/controller/postController.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const createPost = async (req, res) => {
       if(req.file) savePostImage = await uploadToCloud(req.file, res);
 
     const post = await Posts.create({
-      postImage: savePostImage?.secure_url || "https://res.cloudinary.com/da12yf0am/image/upload/v1696850499/pbxwlozt1po8vtbwyabc.jpg",
+      postImage: savePostImage?.secure_url || "https://res.cloudinary.com/da12yf0am/image/upload/v1698131855/z3epyxo5ytlabbvegjiy.jpg",
```
Comment (mansurissa) [inline src/controller/postController.js]: Can you allow a post without an image? Like when no image provided or if Cloudinary returns an error you can a post be crested

File: src/controller/userController.js [hunk 2]
```diff
@@ -157,8 +157,9 @@ export const userUpdate = async (req, res) =>{
         })
         }
       }
-    let updateProfile;
-      const encryptpass = await bcrypt.genSalt(10);
+      let updateProfile;
+      if(password){
+        const encryptpass = await bcrypt.genSalt(10);
```
Comment (mansurissa) [inline src/controller/userController.js]: Enforce camelCase

File: src/controller/userController.js [hunk 3]
```diff
@@ -168,13 +169,28 @@ export const userUpdate = async (req, res) =>{
...
+ 
+    });
+      }
+      else{
+        if(req.file) updateProfile = await uploadToCloud(req.file, res);
+    const userUpdate = await Users.findByIdAndUpdate(id, {
+      firstName,
+      lastName,
+      email,
+      profile: updateProfile?.secure_url || "https://res.cloudinary.com/da12yf0am/image/upload/v1696850499/pbxwlozt1po8vtbwyabc.jpg",
```
Comment (mansurissa) [inline src/controller/userController.js]: Same case for image here.

**Review by mansurissa — CHANGES_REQUESTED** [review]
Work on the comments please

File: src/controller/postController.js (hunk 1 above)
Comment (NP-SON-250) [reply src/controller/postController.js]: Corrected camelCase

File: src/controller/userController.js (hunk 3 above)
Comment (NP-SON-250) [reply src/controller/userController.js]: removed default links for image upload

**Comment by NP-SON-250** [comment]
## What this PR do?
1. Correct request change

**Comment by NP-SON-250** [comment]
## What this PR do?
1. Handle requested changes

**Review by mansurissa — APPROVED** [review]
Good job

**Pull request merged**